File: aerox.Runtime.Widgets/Events/CursorDownEvent.cs
using aerox.Runtime.Math;

namespace aerox.Runtime.Widgets.Events;

public class CursorDownEvent : Event
{
    public Vector2<float> Position;

    public CursorDownEvent(WidgetSurface surface, Vector2<float> position) : base(surface)
    {
        Position = position;
    }
}

File: modules/widgets/lib/aerox/widgets/event/CursorDownEvent.cpp
#include "aerox/widgets/event/CursorDownEvent.hpp"

namespace aerox::widgets
{
    CursorDownEvent::CursorDownEvent(const Shared<Surface>& inSurface, const window::CursorButton inButton,
        const Vec2<float>& inPosition) : Event(inSurface), button(inButton), position(inPosition)
    {
        
    }
}
